src/main/java/carpet/script/utils/ShapeDispatcher.java

```diff
@@ -858,6 +925,33 @@ public long calcKey()
     public static abstract class Param
     {
         public static Map<String, Param> of = new HashMap<String, Param>(){{
+            put("vertex", new PointsParam("vertex"){
```

Comment:
Points param is unused, so you can change it to not use expensive x, y, z fields